lib/tasks/republish.rake

```diff
@@ -0,0 +1,9 @@
+desc "Republish all documents with non-english html attachments"
+task republish_non_english_html_attachments: :environment do
+  HtmlAttachment.where.not(deleted: true)
+  .where.not(locale: "en")
+  .where.not(locale: nil) # HtmlAttachment#sluggable_locale? treats nil locales the same as English
+  .each do |attachment|
+    PublishingApiDocumentRepublishingWorker.perform_async(attachment.attachable.document.id)
```

Comment:
I've just tried this with that example:

```
PublishingApiDocumentRepublishingWorker.perform_async(HtmlAttachment.find(5654420).attachable.document.id)
=> "9c733a97d538128d594d1a2d"
```

...it worked! 🎉 

So now the last thing I want to try is testing what happens when there is a draft edition (i.e. we don't want to accidentally publish something that's in draft).

Reply:
Ah yeah, I think at some point I was filtering based on status but I think that got lost along the way.

(I think because I previously also wanted to filter out the `deleted` state but then found I could check the attachment's `deleted` property and in fact _had_ to check that property. Hopefully it won't require too much to restore the `state` check, though it might need the join to be reintroduced.)

Reply:
Just tried the command above on a HtmlAttachment that was part of a draft publication.

It didn't publish the publication or the attachment 👍